From 166d083ef1ded7066870e2eeaab6b1e9a4f8bcda Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Sat, 12 Jun 2021 13:20:39 -0700
Subject: [PATCH] Update packaged filenames for gitian builds (#193)

Changing the CMake project name updates the filename of the
`ninja package_source` command and the resulting output filenames
of the gitian builds. This commit updates the package name to be in
line with the branding for lotus and to reduce confusion for users.

Rebase provenance: this historical change was adapted onto the pinned
Bitcoin ABC base. Source diff is the recorded Lotus V1 replay commit below.

Historical-source-commit: 8b50aecee3cad40cb4b3c1a531f018cf649cdb00
Recorded-replay-commit: 9c84d0a36d9bbc137ff572f99f2f1da783d1278b

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 72242fd8b1..33a0bed6a3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,10 +6,10 @@ set(CMAKE_USER_MAKE_RULES_OVERRIDE
 	"${CMAKE_SOURCE_DIR}/cmake/modules/OverrideInitFlags.cmake"
 )
 
-project(bitcoin-abc
+project(lotus
 	VERSION 0.34.1
 	DESCRIPTION "Lotus is a full node implementation of the Lotus protocol."
-	HOMEPAGE_URL "https://www.bitcoinabc.org"
+	HOMEPAGE_URL "https://www.givelotus.org"
 )
 
 add_custom_target(print-version
diff --git a/doc/doxygen/Doxyfile.cmake.in b/doc/doxygen/Doxyfile.cmake.in
index 70a5b5594a..244c29c6a4 100644
--- a/doc/doxygen/Doxyfile.cmake.in
+++ b/doc/doxygen/Doxyfile.cmake.in
@@ -38,7 +38,7 @@ PROJECT_NAME           = "${PACKAGE_NAME}"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = ${bitcoin-abc_VERSION}
+PROJECT_NUMBER         = ${lotus_VERSION}
 
 # Using the PROJECT_BRIEF tag one can provide an optional one line description
 # for a project that appears at the top of each page and should give viewer a
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3a961414e3..54e1f1d521 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Copyright (c) 2017 The Bitcoin developers
 
-project(lotusd)
+project(lotus-binaries)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
diff --git a/src/qt/CMakeLists.txt b/src/qt/CMakeLists.txt
index 40837906a6..8167143cd9 100644
--- a/src/qt/CMakeLists.txt
+++ b/src/qt/CMakeLists.txt
@@ -413,12 +413,12 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 		OUTPUT_NAME "${BITCOIN_QT_OSX_BUNDLE_NAME}"
 		MACOSX_BUNDLE_INFO_PLIST "${CMAKE_SOURCE_DIR}/share/qt/Info.plist.cmake.in"
 		MACOSX_BUNDLE_BUNDLE_NAME "${BITCOIN_QT_OSX_BUNDLE_NAME}"
-		MACOSX_BUNDLE_BUNDLE_VERSION "${bitcoin-abc_VERSION}"
+		MACOSX_BUNDLE_BUNDLE_VERSION "${lotus_VERSION}"
 		MACOSX_BUNDLE_GUI_IDENTIFIER "org.lotus.${BITCOIN_QT_OSX_BUNDLE_NAME}"
 		MACOSX_BUNDLE_ICON_FILE "${BITCOINQT_BUNDLE_ICON_NAME}"
-		MACOSX_BUNDLE_INFO_STRING "${bitcoin-abc_VERSION}, Copyright © 2009-${COPYRIGHT_YEAR} ${COPYRIGHT_HOLDERS_FINAL}"
-		MACOSX_BUNDLE_LONG_VERSION_STRING "${bitcoin-abc_VERSION}"
-		MACOSX_BUNDLE_SHORT_VERSION_STRING "${bitcoin-abc_VERSION}"
+		MACOSX_BUNDLE_INFO_STRING "${lotus_VERSION}, Copyright © 2009-${COPYRIGHT_YEAR} ${COPYRIGHT_HOLDERS_FINAL}"
+		MACOSX_BUNDLE_LONG_VERSION_STRING "${lotus_VERSION}"
+		MACOSX_BUNDLE_SHORT_VERSION_STRING "${lotus_VERSION}"
 		RESOURCE "${BITCOINQT_BUNDLE_RESOURCES}"
 	)
 	# The InfoPlist.strings files should be located in a resource subdirectory.
